src/libflake/flake/flake.cc

```diff
@@ -672,6 +672,8 @@ LockedFlake lockFlake(
             if (lockFlags.writeLockFile) {
                 if (sourcePath || lockFlags.outputLockFilePath) {
                     if (auto unlockedInput = newLockFile.isUnlocked()) {
+                        if (lockFlags.failOnUnlocked)
+                            throw Error("cannot write lock file of flake '%s' because it has an unlocked input ('%s').\n", topRef, *unlockedInput);
```

Comment:
```suggestion
                            throw Error("cannot write lock file of flake '%s' because it has an unlocked input ('%s'). Unlocked inputs can be passed to most nix commands using --override-input, but do not result in a lock file.\n", topRef, *unlockedInput);
```

Reply:
I thought about that, but unlocked inputs don't need to come from `--override-input`. So this message could be potentially confusing.

Reply:
In that case, we shouldn't have it.